cypress/support/Projects.js

```diff
@@ -4,647 +4,750 @@ const DEFAULT_TIMEOUT = 60000;
-                deleteProject(projectName);
-            });
+      }
+
+      if (options.systemPrompt) {
+        it('Add system prompt', () => {
+          clickSystemPromptsTab(projectName);
+          uploadFileWithFileBrowser(
+            options.systemPrompt,
+            projectName,
+            'System Prompts'
+          );
         });
-    }
+      }
 
-    static createNotebook(notebookName, projectName) {
-        it('Should create a new notebook', () => {
-            createNotebook(notebookName, projectName);
+      if (options.createNotebook) {
+        it('Create notebook', () => {
+          createNotebook(options.createNotebook, projectName);
         });
-    }
+      }
+    });
+  }
 
-    static shareProject(projectName, notebook, user) {
-        describe(`Logging in to user --${user}`, () => {
-            it(`Should share project: ${projectName}`, () => {
-                logoutUser();
-                loginAs(user);
-                checkInbox(projectName);
-                handleProjectInvite(projectName, "accept");
-                validateSharedProjects(projectName, notebook, user);
-            });
-        });
-    }
+  static createProject(projectName) {
+    it('Should create a new project', () => {
+      // checkAndDeleteProjectIfExists('Renamed Test Project')
+      createProject(projectName);
+    });
+  }
 
-    static systemPromptOperations(projectName, promptName) {
-        describe(`System Prompt Operations for ${projectName}`, () => {
-            it(`Should view system prompt: ${promptName}`, () => {
-                handleSystemPrompt(projectName, 'view', promptName);
-            });
+  static renameProject(oldName, newName) {
+    it('Should rename a project', () => {
+      renameProject(oldName, newName);
+    });
+  }
+
+  static deleteProject(projectName) {
+    describe('Delete Project', () => {
+      it('Should delete a project', () => {
+        logoutUser();
+        loginAs('admin');
+        deleteProject(projectName);
+      });
+    });
+  }
 
-            it(`Should delete system prompt: ${promptName}`, () => {
-                handleSystemPrompt(projectName, 'delete', promptName);
-            });
-        });
-    }
+  static createNotebook(notebookName, projectName) {
+    it('Should create a new notebook', () => {
+      createNotebook(notebookName, projectName);
+    });
+  }
+
+  static shareProject(projectName, notebook, user) {
+    describe(`Logging in to user --${user}`, () => {
+      it(`Should share project: ${projectName}`, () => {
+        cy.clearAllStorage();
+        loginAs(user);
+        checkInbox(projectName);
+        handleProjectInvite(projectName, 'accept');
+        validateSharedProjects(projectName, notebook, user);
+      });
+    });
+  }
+
+  static systemPromptOperations(projectName, promptName) {
+    describe(`System Prompt Operations for ${projectName}`, () => {
+      before(() => {
+        cy.clearAllStorage();
+        loginAs('admin');
```

Comment:
Ensure to login again as admin, since auto-share is not the owner of the project, thus only admin can perform these tasks